activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -115,7 +115,16 @@ def load_records(&block)
             # Make several smaller queries if necessary or make one query if the adapter supports it
             slices = owner_keys.each_slice(klass.connection.in_clause_length || owner_keys.size)
             @preloaded_records = slices.flat_map do |slice|
-              records_for(slice).load(&block)
+              records = records_for(slice)
```

Comment:
Not sure what you are suggesting here, would lead to the same issue as you can not call load on an array